src/streaming_decoder.rs

```diff
@@ -39,10 +39,35 @@ impl<READ: Read> StreamingDecoder<READ, FrameDecoder> {
         decoder.init(&mut source)?;
         Ok(StreamingDecoder { decoder, source })
     }
+}
 
-    pub fn inner(self) -> FrameDecoder {
+impl<READ: Read, DEC: BorrowMut<FrameDecoder>> StreamingDecoder<READ, DEC> {
+    /// Destructures this object into the inner [FrameDecoder].
+    pub fn inner(self) -> DEC {
         self.decoder
     }
+
+    /// Gets a reference to the underlying reader.
+    /// 
+    /// It is inadvisable to directly read from the underlying reader.
+    pub fn get_ref(&self) -> &READ {
```

Comment:
I think this should be called `inner` instead of the current `inner` function

Reply:
The naming of `get_mut` and `get_ref` were largely to keep consistency with other things in the ecosystem:
- [bzip2](https://docs.rs/bzip2/latest/bzip2/read/struct.BzDecoder.html#method.get_ref)
- [flate2](https://docs.rs/flate2/latest/flate2/read/struct.GzDecoder.html#method.get_ref)
- [xz2](https://docs.rs/xz/latest/xz/bufread/struct.XzDecoder.html#method.get_ref)
- [std::io::BufReader](https://docs.rs/xz/latest/xz/bufread/struct.XzDecoder.html#method.get_ref)

I think I avoided making a `into_inner() -> READ` because it would be confusing with `inner()`. But, if we're willing to change the API, we could have that as well?

Reply:
I didn't realize this was a common api. Then let's keep it this way!